lib/Interpreter/InterOp.cpp

```diff
@@ -1829,6 +1830,50 @@ namespace InterOp {
     return 0;
   }
 
+  namespace {
+  
+  std::string GetExecutablePath(const char *Argv0, void *MainAddr) {
+    return llvm::sys::fs::getMainExecutable(Argv0, MainAddr);
+  }
+
+  std::string MakeResourcesPath() {
+    // Dir is bin/ or lib/, depending on where BinaryPath is.
+    void *MainAddr = (void *)(intptr_t)GetExecutablePath;
+    std::string BinaryPath = GetExecutablePath(/*Argv0=*/nullptr, MainAddr);
+
+    // build/tools/clang/unittests/Interpreter/Executable -> build/
+    llvm::StringRef Dir = llvm::sys::path::parent_path(BinaryPath);
+
+    Dir = llvm::sys::path::parent_path(Dir);
+    Dir = llvm::sys::path::parent_path(Dir);
+    Dir = llvm::sys::path::parent_path(Dir);
+    Dir = llvm::sys::path::parent_path(Dir);
+    //Dir = sys::path::parent_path(Dir);
+    llvm::SmallString<128> P(Dir);
+    llvm::sys::path::append(P, llvm::Twine("lib") + CLANG_LIBDIR_SUFFIX,
+                            "clang", CLANG_VERSION_STRING);
+
+    return std::string(P.str());
+  }
+
+  }
+
+  TInterp_t CreateInterpreter() {
+    std::string MainExecutableName =
+      llvm::sys::fs::getMainExecutable(nullptr, nullptr);
+    std::string ResourceDir = MakeResourcesPath();
+    std::vector<const char *> ClingArgv = {"-resource-dir", ResourceDir.c_str(),
+                                           "-std=c++14"};
+    ClingArgv.insert(ClingArgv.begin(), MainExecutableName.c_str());
+    return (TInterp_t) new cling::Interpreter(ClingArgv.size(), &ClingArgv[0]);
+  }
```

Comment:
This part seems available here https://github.com/compiler-research/InterOp/blob/41a32cce4af78c3b1225b90c84515f144055337f/unittests/InterOp/Utils.cpp#L30

Reply:
Yeah, but I think this wont be required in the InterOp library as the binary path that we get is the path to python so the resource directory cannot be set properly.

For example, in my system, when running `cppyy` the `GetExecutablePath` function returns:
```
/usr/bin/python3.10
```

The resources path cannot be found by using this and thus is currently using the wrong directory in the `-resource-dir` argument.